### PR TITLE
33090 extend sensitive post parameter filtering

### PR DESCRIPTION
**Pull request by Oluwayhemisi**
I created this branch to fix the issue of sensitive post parameter filtering not marking the request until the view is executed, that is, filtering cannot be applied to reports  generated by exceptions in the middleware [for ticket](https://code.djangoproject.com/ticket/33090). Carlton Gibson already suggested a solution which I worked on and also wrote a test for it.

### CODE REVIEW

File: django/core/handlers/base.py [hunk 1]
```diff
@@ -179,8 +179,11 @@ def _get_response(self, request):
         """
         response = None
         callback, callback_args, callback_kwargs = self.resolve_request(request)
+        # Mark the request with sensitive_post_parameters if applied.
+        if hasattr(callback, "sensitive_post_parameters"):
+            request.sensitive_post_parameters = callback.sensitive_post_parameters
 
-        # Apply view middleware
+        # Apply view middlewarec;
```
Comment (sarahboyce) [inline django/core/handlers/base.py:186]:
```suggestion
        # Apply view middleware
```

File: django/core/handlers/base.py [hunk 2]
```diff
@@ -234,6 +237,9 @@ async def _get_response_async(self, request):
         response = None
         callback, callback_args, callback_kwargs = self.resolve_request(request)
 
+        # Mark the request with sensitive_post_parameters if applied.
+        if hasattr(callback, "sensitive_post_parameters"):
+            request.sensitive_post_parameters = callback.sensitive_post_parameters
```
Comment (sarahboyce) [inline django/core/handlers/base.py:242]: Can we add a similar test with an async view?

File: django/core/handlers/base.py [hunk 3]
```diff
@@ -179,8 +179,11 @@ def _get_response(self, request):
         """
         response = None
         callback, callback_args, callback_kwargs = self.resolve_request(request)
+        # Mark the request with sensitive_post_parameters if applied.
+        if hasattr(callback, "sensitive_post_parameters"):
+            request.sensitive_post_parameters = callback.sensitive_post_parameters
```
Comment (sarahboyce) [inline django/core/handlers/base.py:184]: Reading the ticket, it sounds like that this is called before the middleware methods is quite important.
Currently if I move this below where middleware is applied, the test still passes.
Can you add something in the test to cover that the request has sensitive_post_parameters set also in the middleware method?

**Review by sarahboyce — COMMENTED** [review]
Thank you for this @Oluwayhemisi !
Looks really good, I appreciate this is in draft so maybe you weren't wanting a review yet but I have a couple of suggestions that might help you get this in :) 

When you've had time to come back to this, please can you unmark the PR as draft and also in the ticket unmark "Patch needs improvement"? Will make sure you're in the review queue :star:

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**